arangod/Transaction/Methods.cpp

```diff
@@ -1062,15 +1061,55 @@ Future<OperationResult> transaction::Methods::insertLocal(std::string const& cna
     TRI_ASSERT(!isPrimaryKeyConstraintViolation || !key.isNone());
 
     bool didReplace = false;
+
+#ifdef USE_ENTERPRISE
```

Comment:
Why does this code exist?
This should all be done via the Schema Validator.
And there should be no late injection of this check, the Schema should only let correct edges pass, and the schema should be identical on SingleServer and Cluster.

Reply:
@hkernbach sorry brain overflow, did we talk about this?

Reply:
But those are additional PRs, got it :+1: